### DISCUSSION TITLE
Error with reading the database

### DISCUSSION THREAD

**Discussion by Maxmkmk** [inbo/camtraptor#305] [Q&A] [answered]
hi, 
I am receiving an error when reading the data exported from Agouti. 
```
pkg <- read_camtrapDP("./datapackage.json")
Error in `dplyr::left_join()`:
! Can't join `x$mediaID` with `y$mediaID` due to incompatible types.
ℹ `x$mediaID` is a <logical>.
ℹ `y$mediaID` is a <character>.
```
```
> rlang::last_trace()
<error/dplyr_error_join_incompatible_type>
Error in `dplyr::left_join()`:
! Can't join `x$mediaID` with `y$mediaID` due to incompatible types.
ℹ `x$mediaID` is a <logical>.
ℹ `y$mediaID` is a <character>.
---
Backtrace:
    ▆
 1. ├─camtrapDensity::read_camtrapDP("./datapackage.json")
 2. │ └─... %>% ...
 3. ├─dplyr::left_join(...)
 4. └─dplyr:::left_join.data.frame(., dplyr::select(media, mediaID, fileName, timestamp), by = "mediaID")
```

Can you please help. Thanks!

**Comment by damianooldoni** [comment]
Hi @Maxmkmk! Thanks for reaching us. I will make an issue of this. I think I know the reason of this bug. We don't define column types explicitly and it seems that we are trying to join internally a dataframe where `mediaID` is not present (NA is logical) with a dataframe where `mediaID` is a character as there are some not NAs.

I will try to check this asap.

Please, also notice that this behavior is very likely due to the down-conversion to 0.1.6 camtraptor is performing while reading camtrapdp packages.

To read camtrap DPs as they are (version 1.0) you can use [camtrapdp](https://inbo.github.io/camtrapdp/index.html) package and its function: [`read_camtrapdp()`](https://inbo.github.io/camtrapdp/reference/read_camtrapdp.html).

In the meantime, we are working to remove `read_camtrap_dp()` from camtraptor and converting all functions to support camtrapDPs version 1. Hopefully camtraptor will be ready before end of May.

**Comment by Maxmkmk** [comment]
Thank you.
I am using an ubuntu system. Might it be the reason?

I was able to pull the database with this command 
`pkg <- read_camtrap_dp("datapackage.json")`
with the reply
`The dataset uses Camtrap DP version 1.0, it has been converted to 0.1.6.                                                                                                            
See https://inbo.github.io/camtraptor/#camtrap-dp for details.`

However, thereafter I receive the same error related to the location 
```
> res <- rem_estimate(pkg)
Error in check_deployment_models(package) : 
  The specified folder does not exist.
```

**Comment by damianooldoni** [comment]
Hi @Maxmkmk. No, I don't think the OS (Ubuntuà is the origin of the problem. 

It would be great if you could share the Camtrap Data Package causing the issue. I will not share it with nobody and I will delete it after solving the issue. Mail to: [damiano.oldoni@inbo.be](mailto:damiano.oldoni@inbo.be).

**Reply by Maxmkmk** [reply]
Thanks a lot for your support.

I  have tried the same package with  a windows OS PC, but the outcome is the same.

I am sending the data package to you.

Thank you.

**Comment by damianooldoni** [comment, marked as answer]
Hi @Maxmkmk! Sorry for making you wait so long. I downloaded your package and tried to import your camtrap Data Package and it works smoothly!
Here below what I get:

```r
library(camtraptor)
# change path where needed 
path_to_json <- "../../2023_data/datapackage.json"
ctdp <- read_camtrap_dp(file = path_to_json)
The dataset uses Camtrap DP version 1.0, it has been converted to 0.1.6.                                               
See https://inbo.github.io/camtraptor/#camtrap-dp for details.
```

While better reading your code in your very first post, I noticed that you are using another function, probably from another package: `read_camtrapDP()`.

Notice that if you don't want to downconvert to Camtrap Data Package version 0.1.6, you can use the new [camtrapdp](https://inbo.github.io/camtrapdp/) package, which reads Agouti packages with version 1.0:

```r
library(camtrapdp)
# change path where needed 
path_to_json <- "../../2023_data/datapackage.json"
ctdp_new_version <- camtrapdp::read_camtrapdp(file = path_to_json)
version(ctdp_new_version)
[1] "1.0"
```

As I told you, we are working to refactor all camtraptor functions to support camtrap DPs version 1.0 as read using [camtrapdp](https://inbo.github.io/camtrapdp/) package.
